src/lib/comms/sol-lwm2m.c

```diff
@@ -4759,6 +5451,14 @@ sol_lwm2m_client_start(struct sol_lwm2m_client *client)
 
     return 0;
 
+err_access_control:
+    if (client->supports_access_control) {
+        ctx = find_object_ctx_by_id(client, ACCESS_CONTROL_OBJECT_ID);
+        while ((instance = sol_vector_get(&ctx->instances, 0)) != NULL) {
+            obj_instance_clear(client, ctx, instance);
+            sol_vector_del_element(&ctx->instances, instance);
+        }
+    }
```

Comment:
All elements should be deleted in there too (it's LWM2M Bootstrap Delete on '/'). But I see your point, I will change these 2 you suggested.